The job **`x86_64-gnu-llvm-15`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/5955321170/job/16153714034) [(plain)](https://github.com/rust-lang/rust/commit/0576f80be752ab0d1c809e7d786d4d3a343bf12e/checks/16153714034/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
GITHUB_ENV=/home/runner/work/_temp/_runner_file_commands/set_env_83e6804d-add3-489b-aab8-b6430cfff687
GITHUB_EVENT_NAME=pull_request
GITHUB_EVENT_PATH=/home/runner/work/_temp/_github_workflow/event.json
GITHUB_GRAPHQL_URL=https://api.github.com/graphql
GITHUB_HEAD_REF=parallel-guard
GITHUB_JOB=pr
GITHUB_PATH=/home/runner/work/_temp/_runner_file_commands/add_path_83e6804d-add3-489b-aab8-b6430cfff687
GITHUB_REF=refs/pull/115144/merge
GITHUB_REF_NAME=115144/merge
GITHUB_REF_PROTECTED=false
---
failures:

---- [ui] tests/ui/const-generics/late-bound-vars/in_closure.rs stdout ----

error: Error: expected failure status (Some(101)) but received status Some(1).
command: RUSTC_ICE="0" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui/const-generics/late-bound-vars/in_closure.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "--sysroot" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2" "--target=x86_64-unknown-linux-gnu" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "--emit" "metadata" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/const-generics/late-bound-vars/in_closure" "-A" "unused" "-A" "internal_features" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/const-generics/late-bound-vars/in_closure/auxiliary"
stdout: none
--- stderr -------------------------------
--- stderr -------------------------------
##[error]error: internal compiler error: compiler/rustc_borrowck/src/universal_regions.rs:880:36: cannot convert `ReFree(DefId(0:5 ~ in_closure[2ba9]::test), BrNamed(DefId(0:6 ~ in_closure[2ba9]::test::'a), 'a))` to a region vid
thread 'rustc' panicked at /checkout/compiler/rustc_errors/src/lib.rs:1635:9:
Box<dyn Any>
stack backtrace:
   0:     0x7f06d1d2829c - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h2c610fcd1d0507f7
   0:     0x7f06d1d2829c - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h2c610fcd1d0507f7
   1:     0x7f06d1d96187 - core::fmt::write::h532c1a9d2529a174
   2:     0x7f06d1d1c485 - std::io::Write::write_fmt::hcca341598a33358e
   3:     0x7f06d1d28085 - std::sys_common::backtrace::print::h916984cc5b632ec5
   4:     0x7f06d1d2b54f - std::panicking::panic_hook_with_disk_dump::{{closure}}::h59108b0e3086fc8a
   5:     0x7f06d1d2b27f - std::panicking::panic_hook_with_disk_dump::h3944a5f656361980
   6:     0x7f06d295fd70 - rustc_driver_impl[2368e4403bee663b]::install_ice_hook::{closure#0}
   7:     0x7f06d1d2bd57 - std::panicking::rust_panic_with_hook::hd3a292a3d874503f
   8:     0x7f06d53c1fb3 - std[c23006139a18bab8]::panicking::begin_panic::<rustc_errors[f5c82d59ba6d4ae9]::ExplicitBug>::{closure#0}
   9:     0x7f06d53b8456 - std[c23006139a18bab8]::sys_common::backtrace::__rust_end_short_backtrace::<std[c23006139a18bab8]::panicking::begin_panic<rustc_errors[f5c82d59ba6d4ae9]::ExplicitBug>::{closure#0}, !>
  10:     0x7f06d28ca7f6 - std[c23006139a18bab8]::panicking::begin_panic::<rustc_errors[f5c82d59ba6d4ae9]::ExplicitBug>
  11:     0x7f06d553c4f7 - <rustc_errors[f5c82d59ba6d4ae9]::HandlerInner>::bug::<alloc[ea73a18a91278db0]::string::String>
  12:     0x7f06d553c359 - <rustc_errors[f5c82d59ba6d4ae9]::Handler>::bug::<alloc[ea73a18a91278db0]::string::String>
  13:     0x7f06d5560a87 - rustc_middle[54977464e61939cc]::util::bug::opt_span_bug_fmt::<rustc_span[96e4dced9bc1a569]::span_encoding::Span>::{closure#0}
  14:     0x7f06d55609bc - rustc_middle[54977464e61939cc]::ty::context::tls::with_opt::<rustc_middle[54977464e61939cc]::util::bug::opt_span_bug_fmt<rustc_span[96e4dced9bc1a569]::span_encoding::Span>::{closure#0}, !>::{closure#0}
  15:     0x7f06d5560954 - rustc_middle[54977464e61939cc]::ty::context::tls::with_context_opt::<rustc_middle[54977464e61939cc]::ty::context::tls::with_opt<rustc_middle[54977464e61939cc]::util::bug::opt_span_bug_fmt<rustc_span[96e4dced9bc1a569]::span_encoding::Span>::{closure#0}, !>::{closure#0}, !>
  16:     0x7f06d28d05e2 - rustc_middle[54977464e61939cc]::util::bug::bug_fmt
Build completed unsuccessfully in 0:10:56
  17:     0x7f06d3eeab7e - <rustc_borrowck[e1deb4dffcf0a0b1]::universal_regions::UniversalRegions>::to_region_vid
  18:     0x7f06d3f94d12 - <rustc_borrowck[e1deb4dffcf0a0b1]::type_check::constraint_conversion::ConstraintConversion>::convert
  19:     0x7f06d3f96047 - <rustc_borrowck[e1deb4dffcf0a0b1]::type_check::constraint_conversion::ConstraintConversion>::convert_all
  20:     0x7f06d3f46256 - <rustc_borrowck[e1deb4dffcf0a0b1]::type_check::TypeChecker>::push_region_constraints
  21:     0x7f06d3f4370d - <rustc_borrowck[e1deb4dffcf0a0b1]::type_check::TypeChecker>::ascribe_user_type
  22:     0x7f06d3f2da61 - rustc_borrowck[e1deb4dffcf0a0b1]::type_check::type_check
  23:     0x7f06d3eba27c - rustc_borrowck[e1deb4dffcf0a0b1]::nll::compute_regions
  24:     0x7f06d3e15172 - rustc_borrowck[e1deb4dffcf0a0b1]::do_mir_borrowck
  25:     0x7f06d3dff91a - rustc_borrowck[e1deb4dffcf0a0b1]::mir_borrowck
  26:     0x7f06d41f7707 - rustc_query_impl[aea78b8e32f0a47c]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[aea78b8e32f0a47c]::query_impl::mir_borrowck::dynamic_query::{closure#2}::{closure#0}, rustc_middle[54977464e61939cc]::query::erase::Erased<[u8; 8usize]>>
  27:     0x7f06d447e799 - <rustc_query_impl[aea78b8e32f0a47c]::query_impl::mir_borrowck::dynamic_query::{closure#2} as core[5f0547cb6bb90da1]::ops::function::FnOnce<(rustc_middle[54977464e61939cc]::ty::context::TyCtxt, rustc_span[96e4dced9bc1a569]::def_id::LocalDefId)>>::call_once
  28:     0x7f06d456f92f - rustc_query_system[cb72d25adf859c92]::query::plumbing::try_execute_query::<rustc_query_impl[aea78b8e32f0a47c]::DynamicConfig<rustc_query_system[cb72d25adf859c92]::query::caches::VecCache<rustc_span[96e4dced9bc1a569]::def_id::LocalDefId, rustc_middle[54977464e61939cc]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[aea78b8e32f0a47c]::plumbing::QueryCtxt, false>
  29:     0x7f06d4372c56 - rustc_query_impl[aea78b8e32f0a47c]::query_impl::mir_borrowck::get_query_non_incr::__rust_end_short_backtrace
  30:     0x7f06d3584946 - rustc_middle[54977464e61939cc]::query::plumbing::query_get_at::<rustc_query_system[cb72d25adf859c92]::query::caches::VecCache<rustc_span[96e4dced9bc1a569]::def_id::LocalDefId, rustc_middle[54977464e61939cc]::query::erase::Erased<[u8; 8usize]>>>
  31:     0x7f06d35a01e6 - rustc_mir_transform[31dc8e217fb5e0d5]::mir_drops_elaborated_and_const_checked
  32:     0x7f06d42154b7 - rustc_query_impl[aea78b8e32f0a47c]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[aea78b8e32f0a47c]::query_impl::mir_drops_elaborated_and_const_checked::dynamic_query::{closure#2}::{closure#0}, rustc_middle[54977464e61939cc]::query::erase::Erased<[u8; 8usize]>>
  33:     0x7f06d43a16e9 - <rustc_query_impl[aea78b8e32f0a47c]::query_impl::mir_drops_elaborated_and_const_checked::dynamic_query::{closure#2} as core[5f0547cb6bb90da1]::ops::function::FnOnce<(rustc_middle[54977464e61939cc]::ty::context::TyCtxt, rustc_span[96e4dced9bc1a569]::def_id::LocalDefId)>>::call_once
  34:     0x7f06d456f92f - rustc_query_system[cb72d25adf859c92]::query::plumbing::try_execute_query::<rustc_query_impl[aea78b8e32f0a47c]::DynamicConfig<rustc_query_system[cb72d25adf859c92]::query::caches::VecCache<rustc_span[96e4dced9bc1a569]::def_id::LocalDefId, rustc_middle[54977464e61939cc]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[aea78b8e32f0a47c]::plumbing::QueryCtxt, false>
  35:     0x7f06d436fcf6 - rustc_query_impl[aea78b8e32f0a47c]::query_impl::mir_drops_elaborated_and_const_checked::get_query_non_incr::__rust_end_short_backtrace
  36:     0x7f06d3584946 - rustc_middle[54977464e61939cc]::query::plumbing::query_get_at::<rustc_query_system[cb72d25adf859c92]::query::caches::VecCache<rustc_span[96e4dced9bc1a569]::def_id::LocalDefId, rustc_middle[54977464e61939cc]::query::erase::Erased<[u8; 8usize]>>>
  37:     0x7f06d359fe1b - rustc_mir_transform[31dc8e217fb5e0d5]::mir_for_ctfe
  38:     0x7f06d41f78f9 - rustc_query_impl[aea78b8e32f0a47c]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[aea78b8e32f0a47c]::query_impl::mir_for_ctfe::dynamic_query::{closure#2}::{closure#0}, rustc_middle[54977464e61939cc]::query::erase::Erased<[u8; 8usize]>>
  39:     0x7f06d4313ded - <rustc_query_impl[aea78b8e32f0a47c]::query_impl::mir_for_ctfe::dynamic_query::{closure#2} as core[5f0547cb6bb90da1]::ops::function::FnOnce<(rustc_middle[54977464e61939cc]::ty::context::TyCtxt, rustc_span[96e4dced9bc1a569]::def_id::DefId)>>::call_once
  40:     0x7f06d452f979 - rustc_query_system[cb72d25adf859c92]::query::plumbing::try_execute_query::<rustc_query_impl[aea78b8e32f0a47c]::DynamicConfig<rustc_query_system[cb72d25adf859c92]::query::caches::DefaultCache<rustc_span[96e4dced9bc1a569]::def_id::DefId, rustc_middle[54977464e61939cc]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[aea78b8e32f0a47c]::plumbing::QueryCtxt, false>
  41:     0x7f06d42c7177 - rustc_query_impl[aea78b8e32f0a47c]::query_impl::mir_for_ctfe::get_query_non_incr::__rust_end_short_backtrace
  42:     0x7f06d374063b - rustc_middle[54977464e61939cc]::query::plumbing::query_get_at::<rustc_query_system[cb72d25adf859c92]::query::caches::DefaultCache<rustc_span[96e4dced9bc1a569]::def_id::DefId, rustc_middle[54977464e61939cc]::query::erase::Erased<[u8; 8usize]>>>
  43:     0x7f06d3746b7f - <rustc_const_eval[5a5335bb5c8c9b3a]::const_eval::machine::CompileTimeInterpreter as rustc_const_eval[5a5335bb5c8c9b3a]::interpret::machine::Machine>::load_mir
  44:     0x7f06d369d081 - <rustc_const_eval[5a5335bb5c8c9b3a]::interpret::eval_context::InterpCx<rustc_const_eval[5a5335bb5c8c9b3a]::const_eval::machine::CompileTimeInterpreter>>::load_mir
  45:     0x7f06d37d8066 - rustc_const_eval[5a5335bb5c8c9b3a]::const_eval::eval_queries::eval_to_allocation_raw_provider
  46:     0x7f06d420ada6 - rustc_query_impl[aea78b8e32f0a47c]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[aea78b8e32f0a47c]::query_impl::eval_to_allocation_raw::dynamic_query::{closure#2}::{closure#0}, rustc_middle[54977464e61939cc]::query::erase::Erased<[u8; 16usize]>>
  47:     0x7f06d431dcde - <rustc_query_impl[aea78b8e32f0a47c]::query_impl::eval_to_allocation_raw::dynamic_query::{closure#2} as core[5f0547cb6bb90da1]::ops::function::FnOnce<(rustc_middle[54977464e61939cc]::ty::context::TyCtxt, rustc_middle[54977464e61939cc]::ty::ParamEnvAnd<rustc_middle[54977464e61939cc]::mir::interpret::GlobalId>)>>::call_once
  48:     0x7f06d44f2f97 - rustc_query_system[cb72d25adf859c92]::query::plumbing::try_execute_query::<rustc_query_impl[aea78b8e32f0a47c]::DynamicConfig<rustc_query_system[cb72d25adf859c92]::query::caches::DefaultCache<rustc_middle[54977464e61939cc]::ty::ParamEnvAnd<rustc_middle[54977464e61939cc]::mir::interpret::GlobalId>, rustc_middle[54977464e61939cc]::query::erase::Erased<[u8; 16usize]>>, false, false, false>, rustc_query_impl[aea78b8e32f0a47c]::plumbing::QueryCtxt, false>
  49:     0x7f06d43a89d5 - rustc_query_impl[aea78b8e32f0a47c]::query_impl::eval_to_allocation_raw::get_query_non_incr::__rust_end_short_backtrace
  50:     0x7f06d37c72bf - rustc_middle[54977464e61939cc]::query::plumbing::query_get_at::<rustc_query_system[cb72d25adf859c92]::query::caches::DefaultCache<rustc_middle[54977464e61939cc]::ty::ParamEnvAnd<rustc_middle[54977464e61939cc]::mir::interpret::GlobalId>, rustc_middle[54977464e61939cc]::query::erase::Erased<[u8; 16usize]>>>
  51:     0x7f06d37d7ba2 - rustc_const_eval[5a5335bb5c8c9b3a]::const_eval::eval_queries::eval_to_allocation_raw_provider
  52:     0x7f06d420ada6 - rustc_query_impl[aea78b8e32f0a47c]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[aea78b8e32f0a47c]::query_impl::eval_to_allocation_raw::dynamic_query::{closure#2}::{closure#0}, rustc_middle[54977464e61939cc]::query::erase::Erased<[u8; 16usize]>>
  53:     0x7f06d431dcde - <rustc_query_impl[aea78b8e32f0a47c]::query_impl::eval_to_allocation_raw::dynamic_query::{closure#2} as core[5f0547cb6bb90da1]::ops::function::FnOnce<(rustc_middle[54977464e61939cc]::ty::context::TyCtxt, rustc_middle[54977464e61939cc]::ty::ParamEnvAnd<rustc_middle[54977464e61939cc]::mir::interpret::GlobalId>)>>::call_once
  54:     0x7f06d44f2f97 - rustc_query_system[cb72d25adf859c92]::query::plumbing::try_execute_query::<rustc_query_impl[aea78b8e32f0a47c]::DynamicConfig<rustc_query_system[cb72d25adf859c92]::query::caches::DefaultCache<rustc_middle[54977464e61939cc]::ty::ParamEnvAnd<rustc_middle[54977464e61939cc]::mir::interpret::GlobalId>, rustc_middle[54977464e61939cc]::query::erase::Erased<[u8; 16usize]>>, false, false, false>, rustc_query_impl[aea78b8e32f0a47c]::plumbing::QueryCtxt, false>
  55:     0x7f06d43a89d5 - rustc_query_impl[aea78b8e32f0a47c]::query_impl::eval_to_allocation_raw::get_query_non_incr::__rust_end_short_backtrace
  56:     0x7f06d3749cec - rustc_const_eval[5a5335bb5c8c9b3a]::const_eval::eval_to_valtree
  57:     0x7f06d37cb710 - <rustc_const_eval[5a5335bb5c8c9b3a]::provide::{closure#0} as core[5f0547cb6bb90da1]::ops::function::FnOnce<(rustc_middle[54977464e61939cc]::ty::context::TyCtxt, rustc_middle[54977464e61939cc]::ty::ParamEnvAnd<rustc_middle[54977464e61939cc]::mir::interpret::GlobalId>)>>::call_once
  58:     0x7f06d41fc566 - rustc_query_impl[aea78b8e32f0a47c]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[aea78b8e32f0a47c]::query_impl::eval_to_valtree::dynamic_query::{closure#2}::{closure#0}, rustc_middle[54977464e61939cc]::query::erase::Erased<[u8; 24usize]>>
  59:     0x7f06d4315f6e - <rustc_query_impl[aea78b8e32f0a47c]::query_impl::eval_to_valtree::dynamic_query::{closure#2} as core[5f0547cb6bb90da1]::ops::function::FnOnce<(rustc_middle[54977464e61939cc]::ty::context::TyCtxt, rustc_middle[54977464e61939cc]::ty::ParamEnvAnd<rustc_middle[54977464e61939cc]::mir::interpret::GlobalId>)>>::call_once
  60:     0x7f06d44f4fdd - rustc_query_system[cb72d25adf859c92]::query::plumbing::try_execute_query::<rustc_query_impl[aea78b8e32f0a47c]::DynamicConfig<rustc_query_system[cb72d25adf859c92]::query::caches::DefaultCache<rustc_middle[54977464e61939cc]::ty::ParamEnvAnd<rustc_middle[54977464e61939cc]::mir::interpret::GlobalId>, rustc_middle[54977464e61939cc]::query::erase::Erased<[u8; 24usize]>>, false, false, false>, rustc_query_impl[aea78b8e32f0a47c]::plumbing::QueryCtxt, false>
  61:     0x7f06d43a8af5 - rustc_query_impl[aea78b8e32f0a47c]::query_impl::eval_to_valtree::get_query_non_incr::__rust_end_short_backtrace
  62:     0x7f06d53a7a62 - rustc_middle[54977464e61939cc]::query::plumbing::query_get_at::<rustc_query_system[cb72d25adf859c92]::query::caches::DefaultCache<rustc_middle[54977464e61939cc]::ty::ParamEnvAnd<rustc_middle[54977464e61939cc]::mir::interpret::GlobalId>, rustc_middle[54977464e61939cc]::query::erase::Erased<[u8; 24usize]>>>
  63:     0x7f06d53aab93 - <rustc_middle[54977464e61939cc]::ty::context::TyCtxt>::const_eval_global_id_for_typeck
  64:     0x7f06d53a9aad - <rustc_middle[54977464e61939cc]::ty::context::TyCtxt>::const_eval_resolve_for_typeck
  65:     0x7f06d2ec1a29 - <rustc_middle[54977464e61939cc]::ty::consts::Const>::eval
  66:     0x7f06d2f512a3 - <rustc_hir_typeck[64888d3a1342c42b]::fn_ctxt::FnCtxt>::check_expr_kind
  67:     0x7f06d2edf3ef - <rustc_hir_typeck[64888d3a1342c42b]::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args
  68:     0x7f06d2f503ea - <rustc_hir_typeck[64888d3a1342c42b]::fn_ctxt::FnCtxt>::check_expr_with_expectation
  69:     0x7f06d2f05bd3 - <rustc_hir_typeck[64888d3a1342c42b]::fn_ctxt::FnCtxt>::check_decl_initializer
  70:     0x7f06d2f05dbf - <rustc_hir_typeck[64888d3a1342c42b]::fn_ctxt::FnCtxt>::check_decl
  71:     0x7f06d2f0609b - <rustc_hir_typeck[64888d3a1342c42b]::fn_ctxt::FnCtxt>::check_stmt
  72:     0x7f06d2f066da - <rustc_hir_typeck[64888d3a1342c42b]::fn_ctxt::FnCtxt>::check_block_with_expected
  73:     0x7f06d2f50cbd - <rustc_hir_typeck[64888d3a1342c42b]::fn_ctxt::FnCtxt>::check_expr_kind
  74:     0x7f06d2edf3ef - <rustc_hir_typeck[64888d3a1342c42b]::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args
  75:     0x7f06d2f503ea - <rustc_hir_typeck[64888d3a1342c42b]::fn_ctxt::FnCtxt>::check_expr_with_expectation
  76:     0x7f06d2ee0b12 - <rustc_hir_typeck[64888d3a1342c42b]::fn_ctxt::FnCtxt>::check_return_expr
  77:     0x7f06d3095cbc - rustc_hir_typeck[64888d3a1342c42b]::check::check_fn
  78:     0x7f06d2f48e0d - <rustc_hir_typeck[64888d3a1342c42b]::fn_ctxt::FnCtxt>::check_expr_closure
  79:     0x7f06d2f5103d - <rustc_hir_typeck[64888d3a1342c42b]::fn_ctxt::FnCtxt>::check_expr_kind
  80:     0x7f06d2edf3ef - <rustc_hir_typeck[64888d3a1342c42b]::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args
  81:     0x7f06d2f503ea - <rustc_hir_typeck[64888d3a1342c42b]::fn_ctxt::FnCtxt>::check_expr_with_expectation
  82:     0x7f06d2f05bd3 - <rustc_hir_typeck[64888d3a1342c42b]::fn_ctxt::FnCtxt>::check_decl_initializer
  83:     0x7f06d2f05dbf - <rustc_hir_typeck[64888d3a1342c42b]::fn_ctxt::FnCtxt>::check_decl
  84:     0x7f06d2f0609b - <rustc_hir_typeck[64888d3a1342c42b]::fn_ctxt::FnCtxt>::check_stmt
  85:     0x7f06d2f066da - <rustc_hir_typeck[64888d3a1342c42b]::fn_ctxt::FnCtxt>::check_block_with_expected
  86:     0x7f06d2f50cbd - <rustc_hir_typeck[64888d3a1342c42b]::fn_ctxt::FnCtxt>::check_expr_kind
  87:     0x7f06d2edf3ef - <rustc_hir_typeck[64888d3a1342c42b]::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args
  88:     0x7f06d2f503ea - <rustc_hir_typeck[64888d3a1342c42b]::fn_ctxt::FnCtxt>::check_expr_with_expectation
  89:     0x7f06d2ee0b12 - <rustc_hir_typeck[64888d3a1342c42b]::fn_ctxt::FnCtxt>::check_return_expr
  90:     0x7f06d3095cbc - rustc_hir_typeck[64888d3a1342c42b]::check::check_fn
  91:     0x7f06d302c19a - rustc_hir_typeck[64888d3a1342c42b]::typeck
  92:     0x7f06d4217257 - rustc_query_impl[aea78b8e32f0a47c]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[aea78b8e32f0a47c]::query_impl::typeck::dynamic_query::{closure#2}::{closure#0}, rustc_middle[54977464e61939cc]::query::erase::Erased<[u8; 8usize]>>
  93:     0x7f06d43a1f49 - <rustc_query_impl[aea78b8e32f0a47c]::query_impl::typeck::dynamic_query::{closure#2} as core[5f0547cb6bb90da1]::ops::function::FnOnce<(rustc_middle[54977464e61939cc]::ty::context::TyCtxt, rustc_span[96e4dced9bc1a569]::def_id::LocalDefId)>>::call_once
  94:     0x7f06d456f92f - rustc_query_system[cb72d25adf859c92]::query::plumbing::try_execute_query::<rustc_query_impl[aea78b8e32f0a47c]::DynamicConfig<rustc_query_system[cb72d25adf859c92]::query::caches::VecCache<rustc_span[96e4dced9bc1a569]::def_id::LocalDefId, rustc_middle[54977464e61939cc]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[aea78b8e32f0a47c]::plumbing::QueryCtxt, false>
  95:     0x7f06d4372726 - rustc_query_impl[aea78b8e32f0a47c]::query_impl::typeck::get_query_non_incr::__rust_end_short_backtrace
  96:     0x7f06d335a9cd - <core[5f0547cb6bb90da1]::panic::unwind_safe::AssertUnwindSafe<rustc_data_structures[639f732e6f2cb10]::sync::par_for_each_in<&[rustc_span[96e4dced9bc1a569]::def_id::LocalDefId], <rustc_middle[54977464e61939cc]::hir::map::Map>::par_body_owners<rustc_hir_analysis[be4775f9a9ff19e8]::check_crate::{closure#7}>::{closure#0}>::{closure#0}::{closure#0}> as core[5f0547cb6bb90da1]::ops::function::FnOnce<()>>::call_once
  97:     0x7f06d33aeef9 - std[c23006139a18bab8]::panicking::try::<(), core[5f0547cb6bb90da1]::panic::unwind_safe::AssertUnwindSafe<rustc_data_structures[639f732e6f2cb10]::sync::par_for_each_in<&[rustc_span[96e4dced9bc1a569]::def_id::LocalDefId], <rustc_middle[54977464e61939cc]::hir::map::Map>::par_body_owners<rustc_hir_analysis[be4775f9a9ff19e8]::check_crate::{closure#7}>::{closure#0}>::{closure#0}::{closure#0}>>
  98:     0x7f06d33550c9 - rustc_data_structures[639f732e6f2cb10]::sync::par_for_each_in::<&[rustc_span[96e4dced9bc1a569]::def_id::LocalDefId], <rustc_middle[54977464e61939cc]::hir::map::Map>::par_body_owners<rustc_hir_analysis[be4775f9a9ff19e8]::check_crate::{closure#7}>::{closure#0}>
  99:     0x7f06d31a729b - rustc_hir_analysis[be4775f9a9ff19e8]::check_crate
 100:     0x7f06d2ae27b4 - rustc_interface[3713d81d1c0b1dcb]::passes::analysis
 101:     0x7f06d4217835 - rustc_query_impl[aea78b8e32f0a47c]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[aea78b8e32f0a47c]::query_impl::analysis::dynamic_query::{closure#2}::{closure#0}, rustc_middle[54977464e61939cc]::query::erase::Erased<[u8; 1usize]>>
 102:     0x7f06d436cad5 - <rustc_query_impl[aea78b8e32f0a47c]::query_impl::analysis::dynamic_query::{closure#2} as core[5f0547cb6bb90da1]::ops::function::FnOnce<(rustc_middle[54977464e61939cc]::ty::context::TyCtxt, ())>>::call_once
 103:     0x7f06d44e7a5d - rustc_query_system[cb72d25adf859c92]::query::plumbing::try_execute_query::<rustc_query_impl[aea78b8e32f0a47c]::DynamicConfig<rustc_query_system[cb72d25adf859c92]::query::caches::SingleCache<rustc_middle[54977464e61939cc]::query::erase::Erased<[u8; 1usize]>>, false, false, false>, rustc_query_impl[aea78b8e32f0a47c]::plumbing::QueryCtxt, false>
 104:     0x7f06d43ca32d - rustc_query_impl[aea78b8e32f0a47c]::query_impl::analysis::get_query_non_incr::__rust_end_short_backtrace
 105:     0x7f06d2961d3c - <rustc_middle[54977464e61939cc]::ty::context::GlobalCtxt>::enter::<rustc_driver_impl[2368e4403bee663b]::run_compiler::{closure#1}::{closure#2}::{closure#6}, core[5f0547cb6bb90da1]::result::Result<(), rustc_span[96e4dced9bc1a569]::ErrorGuaranteed>>
 106:     0x7f06d298331c - <scoped_tls[158fd5a1b0b0daea]::ScopedKey<rustc_span[96e4dced9bc1a569]::SessionGlobals>>::set::<rustc_interface[3713d81d1c0b1dcb]::interface::run_compiler<core[5f0547cb6bb90da1]::result::Result<(), rustc_span[96e4dced9bc1a569]::ErrorGuaranteed>, rustc_driver_impl[2368e4403bee663b]::run_compiler::{closure#1}>::{closure#0}, core[5f0547cb6bb90da1]::result::Result<(), rustc_span[96e4dced9bc1a569]::ErrorGuaranteed>>
 107:     0x7f06d297d9b7 - std[c23006139a18bab8]::sys_common::backtrace::__rust_begin_short_backtrace::<rustc_interface[3713d81d1c0b1dcb]::util::run_in_thread_pool_with_globals<rustc_interface[3713d81d1c0b1dcb]::interface::run_compiler<core[5f0547cb6bb90da1]::result::Result<(), rustc_span[96e4dced9bc1a569]::ErrorGuaranteed>, rustc_driver_impl[2368e4403bee663b]::run_compiler::{closure#1}>::{closure#0}, core[5f0547cb6bb90da1]::result::Result<(), rustc_span[96e4dced9bc1a569]::ErrorGuaranteed>>::{closure#0}::{closure#0}, core[5f0547cb6bb90da1]::result::Result<(), rustc_span[96e4dced9bc1a569]::ErrorGuaranteed>>
 108:     0x7f06d29b5d6e - std[c23006139a18bab8]::panicking::try::<core[5f0547cb6bb90da1]::result::Result<(), rustc_span[96e4dced9bc1a569]::ErrorGuaranteed>, core[5f0547cb6bb90da1]::panic::unwind_safe::AssertUnwindSafe<<std[c23006139a18bab8]::thread::Builder>::spawn_unchecked_<rustc_interface[3713d81d1c0b1dcb]::util::run_in_thread_pool_with_globals<rustc_interface[3713d81d1c0b1dcb]::interface::run_compiler<core[5f0547cb6bb90da1]::result::Result<(), rustc_span[96e4dced9bc1a569]::ErrorGuaranteed>, rustc_driver_impl[2368e4403bee663b]::run_compiler::{closure#1}>::{closure#0}, core[5f0547cb6bb90da1]::result::Result<(), rustc_span[96e4dced9bc1a569]::ErrorGuaranteed>>::{closure#0}::{closure#0}, core[5f0547cb6bb90da1]::result::Result<(), rustc_span[96e4dced9bc1a569]::ErrorGuaranteed>>::{closure#1}::{closure#0}>>
 109:     0x7f06d297f0f0 - <<std[c23006139a18bab8]::thread::Builder>::spawn_unchecked_<rustc_interface[3713d81d1c0b1dcb]::util::run_in_thread_pool_with_globals<rustc_interface[3713d81d1c0b1dcb]::interface::run_compiler<core[5f0547cb6bb90da1]::result::Result<(), rustc_span[96e4dced9bc1a569]::ErrorGuaranteed>, rustc_driver_impl[2368e4403bee663b]::run_compiler::{closure#1}>::{closure#0}, core[5f0547cb6bb90da1]::result::Result<(), rustc_span[96e4dced9bc1a569]::ErrorGuaranteed>>::{closure#0}::{closure#0}, core[5f0547cb6bb90da1]::result::Result<(), rustc_span[96e4dced9bc1a569]::ErrorGuaranteed>>::{closure#1} as core[5f0547cb6bb90da1]::ops::function::FnOnce<()>>::call_once::{shim:vtable#0}
 111:     0x7f06d1ad1b43 - <unknown>
 112:     0x7f06d1b63a00 - <unknown>
 113:                0x0 - <unknown>


note: we would appreciate a bug report: https://github.com/rust-lang/rust/issues/new?labels=C-bug%2C+I-ICE%2C+T-compiler&template=ice.md

note: rustc 1.74.0-nightly (b24b9e8f5 2023-08-23) running on x86_64-unknown-linux-gnu

note: compiler flags: -Z threads=1 -Z simulate-remapped-rust-src-base=/rustc/FAKE_PREFIX -Z translate-remapped-path-to-local-path=no -C codegen-units=1 -Z ui-testing -Z deduplicate-diagnostics=no -Z write-long-types-to-disk=no -C strip=debuginfo -C prefer-dynamic -C rpath -C debuginfo=0
query stack during panic:
query stack during panic:
#0 [mir_borrowck] borrow-checking `test::{closure#0}::{constant#1}`
#1 [mir_drops_elaborated_and_const_checked] elaborating drops for `test::{closure#0}::{constant#1}`
#2 [mir_for_ctfe] caching mir of `test::{closure#0}::{constant#1}` for CTFE
#3 [eval_to_allocation_raw] const-evaluating + checking `test::{closure#0}::{constant#1}`
#4 [eval_to_allocation_raw] const-evaluating + checking `test::{closure#0}::{constant#1}`
#5 [eval_to_valtree] evaluating type-level constant
#6 [typeck] type-checking `test`
end of query stack
error: aborting due to previous error
------------------------------------------

```

</details>

--- a/compiler/rustc_data_structures/src/sync.rs
+++ b/compiler/rustc_data_structures/src/sync.rs
@@ -203,7 +203,7 @@ cfg_if! {
 
         #[macro_export]
         macro_rules! parallel {
-            ($($blocks:block),*) => {
+            ($($blocks:block),*) => {{
                 // We catch panics here ensuring that all the blocks execute.
                 // This makes behavior consistent with the parallel compiler.
                 let mut panic = None;
@@ -219,7 +219,7 @@ cfg_if! {
                 if let Some(panic) = panic {
                     ::std::panic::resume_unwind(panic);
                 }
-            }
+            }}
         }
 
         pub fn par_for_each_in<T: IntoIterator>(t: T, mut for_each: impl FnMut(T::Item) + Sync + Send) {

--- a/compiler/rustc_data_structures/src/sync.rs
+++ b/compiler/rustc_data_structures/src/sync.rs
@@ -41,6 +41,7 @@
 //! [^2] `MTLockRef` is a typedef.
 
 pub use crate::marker::*;
+use std::any::Any;
 use std::collections::HashMap;
 use std::hash::{BuildHasher, Hash};
 use std::ops::{Deref, DerefMut};
@@ -102,6 +103,37 @@ mod mode {
 }
 
 pub use mode::{is_dyn_thread_safe, set_dyn_thread_safe_mode};
+
+/// A guard used to hold panics that occur during a parallel section to later by unwound.
+/// This is used for the parallel compiler to prevent fatal errors from non-deterministically
+/// hiding errors by ensuring that everything in the section has completed executing before
+/// continuing with unwinding. It's also used for the non-parallel code to ensure error message
+/// output match the parallel compiler for testing purposes.
+pub struct ParallelGuard {
+    panic: Lock<Option<Box<dyn Any + std::marker::Send + 'static>>>,
+}
+
+impl ParallelGuard {
+    #[inline]
+    pub fn new() -> Self {
+        ParallelGuard { panic: Lock::new(None) }
+    }
+
+    pub fn run<R>(&self, f: impl FnOnce() -> R) -> Option<R> {
+        catch_unwind(AssertUnwindSafe(f))
+            .map_err(|err| {
+                *self.panic.lock() = Some(err);
+            })
+            .ok()
+    }
+
+    #[inline]
+    pub fn unwind(self) {
+        if let Some(panic) = self.panic.into_inner() {
+            resume_unwind(panic);
+        }
+    }
+}
 
 cfg_if! {
     if #[cfg(not(parallel_compiler))] {
@@ -198,66 +230,37 @@ cfg_if! {
             where A: FnOnce() -> RA,
                   B: FnOnce() -> RB
         {
-            (oper_a(), oper_b())
+            let guard = ParallelGuard::new();
+            let a = guard.run(oper_a);
+            let b = guard.run(oper_b);
+            guard.unwind();
+            (a.unwrap(), b.unwrap())
         }
 
         #[macro_export]
         macro_rules! parallel {
             ($($blocks:block),*) => {{
-                // We catch panics here ensuring that all the blocks execute.
-                // This makes behavior consistent with the parallel compiler.
-                let mut panic = None;
-                $(
-                    if let Err(p) = ::std::panic::catch_unwind(
-                        ::std::panic::AssertUnwindSafe(|| $blocks)
-                    ) {
-                        if panic.is_none() {
-                            panic = Some(p);
-                        }
-                    }
-                )*
-                if let Some(panic) = panic {
-                    ::std::panic::resume_unwind(panic);
-                }
+                let mut guard = $crate::sync::ParallelGuard::new();
+                $(guard.run(|| $blocks);)*
+                guard.unwind();
             }}
         }
 
         pub fn par_for_each_in<T: IntoIterator>(t: T, mut for_each: impl FnMut(T::Item) + Sync + Send) {
-            // We catch panics here ensuring that all the loop iterations execute.
-            // This makes behavior consistent with the parallel compiler.
-            let mut panic = None;
+            let guard = ParallelGuard::new();
             t.into_iter().for_each(|i| {
-                if let Err(p) = catch_unwind(AssertUnwindSafe(|| for_each(i))) {
-                    if panic.is_none() {
-                        panic = Some(p);
-                    }
-                }
+                guard.run(|| for_each(i));
             });
-            if let Some(panic) = panic {
-                resume_unwind(panic);
-            }
+            guard.unwind();
         }
 
         pub fn par_map<T: IntoIterator, R, C: FromIterator<R>>(
             t: T,
             mut map: impl FnMut(<<T as IntoIterator>::IntoIter as Iterator>::Item) -> R,
         ) -> C {
-            // We catch panics here ensuring that all the loop iterations execute.
-            let mut panic = None;
-            let r = t.into_iter().filter_map(|i| {
-                match catch_unwind(AssertUnwindSafe(|| map(i))) {
-                    Ok(r) => Some(r),
-                    Err(p) => {
-                        if panic.is_none() {
-                            panic = Some(p);
-                        }
-                        None
-                    }
-                }
-            }).collect();
-            if let Some(panic) = panic {
-                resume_unwind(panic);
-            }
+            let guard = ParallelGuard::new();
+            let r = t.into_iter().filter_map(|i| guard.run(|| map(i))).collect();
+            guard.unwind();
             r
         }
 
@@ -380,7 +383,11 @@ cfg_if! {
                 let (a, b) = rayon::join(move || FromDyn::from(oper_a.into_inner()()), move || FromDyn::from(oper_b.into_inner()()));
                 (a.into_inner(), b.into_inner())
             } else {
-                (oper_a(), oper_b())
+                let guard = ParallelGuard::new();
+                let a = guard.run(oper_a);
+                let b = guard.run(oper_b);
+                guard.unwind();
+                (a.unwrap(), b.unwrap())
             }
         }
 
@@ -415,28 +422,10 @@ cfg_if! {
                     // of a single threaded rustc.
                     parallel!(impl $fblock [] [$($blocks),*]);
                 } else {
-                    // We catch panics here ensuring that all the blocks execute.
-                    // This makes behavior consistent with the parallel compiler.
-                    let mut panic = None;
-                    if let Err(p) = ::std::panic::catch_unwind(
-                        ::std::panic::AssertUnwindSafe(|| $fblock)
-                    ) {
-                        if panic.is_none() {
-                            panic = Some(p);
-                        }
-                    }
-                    $(
-                        if let Err(p) = ::std::panic::catch_unwind(
-                            ::std::panic::AssertUnwindSafe(|| $blocks)
-                        ) {
-                            if panic.is_none() {
-                                panic = Some(p);
-                            }
-                        }
-                    )*
-                    if let Some(panic) = panic {
-                        ::std::panic::resume_unwind(panic);
-                    }
+                    let guard = $crate::sync::ParallelGuard::new();
+                    guard.run(|| $fblock);
+                    $(guard.run(|| $blocks);)*
+                    guard.unwind();
                 }
             };
         }
@@ -449,31 +438,17 @@ cfg_if! {
         ) {
             if mode::is_dyn_thread_safe() {
                 let for_each = FromDyn::from(for_each);
-                let panic: Mutex<Option<_>> = Mutex::new(None);
-                t.into_par_iter().for_each(|i| if let Err(p) = catch_unwind(AssertUnwindSafe(|| for_each(i))) {
-                    let mut l = panic.lock();
-                    if l.is_none() {
-                        *l = Some(p)
-                    }
+                let guard = ParallelGuard::new();
+                t.into_par_iter().for_each(|i| {
+                    guard.run(|| for_each(i));
                 });
-
-                if let Some(panic) = panic.into_inner() {
-                    resume_unwind(panic);
-                }
+                guard.unwind();
             } else {
-                // We catch panics here ensuring that all the loop iterations execute.
-                // This makes behavior consistent with the parallel compiler.
-                let mut panic = None;
+                let guard = ParallelGuard::new();
                 t.into_iter().for_each(|i| {
-                    if let Err(p) = catch_unwind(AssertUnwindSafe(|| for_each(i))) {
-                        if panic.is_none() {
-                            panic = Some(p);
-                        }
-                    }
+                    guard.run(|| for_each(i));
                 });
-                if let Some(panic) = panic {
-                    resume_unwind(panic);
-                }
+                guard.unwind();
             }
         }
 
@@ -487,43 +462,15 @@ cfg_if! {
             map: impl Fn(I) -> R + DynSync + DynSend
         ) -> C {
             if mode::is_dyn_thread_safe() {
-                let panic: Mutex<Option<_>> = Mutex::new(None);
                 let map = FromDyn::from(map);
-                // We catch panics here ensuring that all the loop iterations execute.
-                let r = t.into_par_iter().filter_map(|i| {
-                    match catch_unwind(AssertUnwindSafe(|| map(i))) {
-                        Ok(r) => Some(r),
-                        Err(p) => {
-                            let mut l = panic.lock();
-                            if l.is_none() {
-                                *l = Some(p);
-                            }
-                            None
-                        },
-                    }
-                }).collect();
-
-                if let Some(panic) = panic.into_inner() {
-                    resume_unwind(panic);
-                }
+                let guard = ParallelGuard::new();
+                let r = t.into_par_iter().filter_map(|i| guard.run(|| map(i))).collect();
+                guard.unwind();
                 r
             } else {
-                // We catch panics here ensuring that all the loop iterations execute.
-                let mut panic = None;
-                let r = t.into_iter().filter_map(|i| {
-                    match catch_unwind(AssertUnwindSafe(|| map(i))) {
-                        Ok(r) => Some(r),
-                        Err(p) => {
-                            if panic.is_none() {
-                                panic = Some(p);
-                            }
-                            None
-                        }
-                    }
-                }).collect();
-                if let Some(panic) = panic {
-                    resume_unwind(panic);
-                }
+                let guard = ParallelGuard::new();
+                let r = t.into_iter().filter_map(|i| guard.run(|| map(i))).collect();
+                guard.unwind();
                 r
             }
         }

--- a/tests/ui/const-generics/late-bound-vars/in_closure.rs
+++ b/tests/ui/const-generics/late-bound-vars/in_closure.rs
@@ -1,4 +1,4 @@
-// failure-status: 101
+// failure-status: 1
 // known-bug: unknown
 // error-pattern:internal compiler error
 // normalize-stderr-test "internal compiler error.*" -> ""
@@ -22,7 +22,10 @@
 #![feature(generic_const_exprs)]
 #![allow(incomplete_features)]
 
-const fn inner<'a>() -> usize where &'a (): Sized {
+const fn inner<'a>() -> usize
+where
+    &'a (): Sized,
+{
     3
 }
 

